soc/silabs/silabs_siwx91x/siwg917/nwp.h

```diff
@@ -9,17 +9,22 @@
 
 #define SIWX91X_INTERFACE_MASK (0x03)
 
+typedef struct siwx91x_boot_config {
+	uint8_t oper_mode;
+	bool hidden_ssid;
+	uint8_t max_num_sta;
+} siwx91x_boot_config_feature_t;
```

Comment:
Unless you plan to add more parameters, I am not sure we need to introduce this struct. You can pass the 3 parameters to `siwx91x_nwp_mode_switch()`.